tests/components/brunt/test_config_flow.py

```diff
@@ -0,0 +1,157 @@
+    """Test we get the form."""
+    result = await _flow_submit(hass, data=None)
+    assert result["type"] == "form"
+    assert result["errors"] is None
+
+    with patch(
+        "homeassistant.components.brunt.config_flow.validate_input", return_value=None
+    ), patch(
+        "homeassistant.components.brunt.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            CONF,
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "test-username"
+    assert result2["data"] == CONF
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import(hass):
+    """Test we get the form."""
+
+    with patch(
+        "homeassistant.components.brunt.config_flow.validate_input", return_value=None
+    ), patch(
+        "homeassistant.components.brunt.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await _flow_submit(hass, {"source": config_entries.SOURCE_IMPORT})
+        await hass.async_block_till_done()
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "test-username"
+    assert result["data"] == CONF
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_duplicate_login(hass):
+    """Test uniqueness of username."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONF,
+        title="test-username",
+        unique_id="test-username",
+    )
+    entry.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.brunt.config_flow.validate_input", return_value=None
+    ), patch(
+        "homeassistant.components.brunt.async_setup_entry",
+        return_value=True,
+    ):
+        result = await _flow_submit(hass, {"source": config_entries.SOURCE_IMPORT})
+        await hass.async_block_till_done()
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured_account"
+
+
+async def test_form_duplicate_login(hass):
+    """Test uniqueness of username."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CONF,
+        title="test-username",
+        unique_id="test-username",
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.brunt.config_flow.validate_input", return_value=None
+    ):
+        result = await _flow_submit(hass)
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+@pytest.mark.parametrize("side_effect", ["cannot_connect", "invalid_auth", "unknown"])
+async def test_form_error(hass, side_effect):
+    """Test we handle cannot connect."""
+    with patch(
+        "homeassistant.components.brunt.config_flow.validate_input",
```

Comment:
Patch the library with the appropriate exception. We need two parameters. One for the exception and one for the error string.

Reply:
I think this is now correct, but please check @MartinHjelmare